x-pack/packages/ml/local_storage/src/storage_context.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, {
+  type PropsWithChildren,
+  useEffect,
+  useMemo,
+  useCallback,
+  useState,
+  useContext,
+} from 'react';
+import { omit } from 'lodash';
+
+import type { Storage } from '@kbn/kibana-utils-plugin/public';
+import { isDefined } from '@kbn/ml-is-defined';
+
+interface StorageDefinition {
+  [key: string]: unknown;
+}
+
+type TStorage = Partial<StorageDefinition> | null;
+type TStorageKey = keyof Exclude<TStorage, null>;
+type TStorageMapped<T extends TStorageKey> = T extends string ? unknown : null;
+
+interface StorageAPI {
+  value: TStorage;
+  setValue: <K extends TStorageKey, T extends TStorageMapped<K>>(key: K, value: T) => void;
+  removeValue: <K extends TStorageKey>(key: K) => void;
+}
+
+export function isStorageKey<T>(key: unknown, storageKeys: readonly T[]): key is T {
+  return storageKeys.includes(key as T);
+}
+
+export const MlStorageContext = React.createContext<StorageAPI>({
```

Comment:
Could you add a comment to this to keep the 'public APIs without comments' count at 0 for this package.